src/Spaces/spectralelement.jl

```diff
@@ -412,3 +415,4 @@ function slab(space::AbstractSpectralElementSpace, v, h)
         slab(space.local_geometry, v, h),
     )
 end
+slab(space::AbstractSpectralElementSpace, h) = slab(space, 1, h)
```

Comment:
@jakebolewski how easy is it to extend this function to an array of Ints, `h[ ]`, representing an array of indices of horizontal elements we want to extract our field on?

Reply:
if you need a arb strided view it is probably better to make a copy of the slabs you need into a continuous data layout

Reply:
It's not a stride (contiguous indices) view. The use case I was thinking in particular of is with the return type of the new interface I added, `neighboring_elements`, which returns an array of integer indices. I can loop/traverse them, and for each apply the `slab` method, but if it was easy to extend, we could apply the slab to the entire set of elements

Reply:
it would be easy but then it would have to allocate, in general we take the view that the user allocates and then the processing routine fills in the data.  this makes it ex. easy to re-use a datalayout buffer in your neighboring element example.

Reply:
for your example, it would be better I think to create a datalayout buffer of 8 elements, loop over the neighboring element indices copying the slabs into the pre-alloced buffer, do the processing and then move onto the next element in the optimization routine.  That way even with parallelism the number of buffers is low Nthreads << Nelements so the amount of allocations is reduced.